action/protocol/execution/evm/evm.go

```diff
@@ -418,6 +421,18 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
 		// After EIP-3529: refunds are capped to gasUsed / 5
 		refund = (evmParams.gas - remainingGas) / params.RefundQuotientEIP3529
 	}
+	// adjust refund due to dynamicGas
+	var (
+		refundLastSnapshot = stateDB.RefundLastSnapshot()
+		currentRefund      = stateDB.GetRefund()
+	)
+	if evmErr != nil && !g.IsOkhotsk(blockHeight) && refundLastSnapshot > 0 && refundLastSnapshot != currentRefund {
```

Comment:
even though, genesis and blockheight are both available here, it is better to use `featureCtx.CorrectGasRefund`.

Reply:
same